dex/token.go

```diff
@@ -0,0 +1,68 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package dex
+
+// Gases is a the gas required for various operations.
+type Gases struct {
```

Comment:
Unlike with the ETH swap contract where these values could only change due to a consensus rules change, these gas values could change if the developers of the ERC20 token change their implementation. The implementation can be changed by using a proxy delegate:
https://fravoll.github.io/solidity-patterns/proxy_delegate.html

It would be safer to call `EstimateGas` each time instead of using pre-calculated values.

Reply:
Hmm. Good point. I think it's good to estimate initiations at order time. I was actually considering doing an estimate for all swaps in `FundOrder` anyway (could replace the balance check, I think). We won't be able to estimate redeems and refunds at order time, so we'll need some estimates there.

Reply:
Yeah for redeem and refund we can't do estimates. If there is a nice buffer on top of the estimate, it's unlikely we'll run into any issues.